relay-server/src/actors/global_config.rs

```diff
@@ -0,0 +1,242 @@
+type UpstreamQueryResult =
+    Result<Result<GetGlobalConfigResponse, UpstreamRequestError>, relay_system::SendError>;
+
+/// The response of a fetch of a global config from upstream.
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetGlobalConfigResponse {
+    #[serde(default)]
+    global: Option<GlobalConfig>,
+}
+
+/// The request to fetch a global config from upstream.
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetGlobalConfig {
+    pub global: bool,
+    // Dummy variable - upstream expects a list of public keys.
+    public_keys: Vec<()>,
+}
+
+impl GetGlobalConfig {
+    pub fn query() -> GetGlobalConfig {
+        GetGlobalConfig {
+            global: true,
+            public_keys: vec![],
+        }
+    }
+}
+
+impl UpstreamQuery for GetGlobalConfig {
+    type Response = GetGlobalConfigResponse;
+
+    fn method(&self) -> reqwest::Method {
+        Method::POST
+    }
+
+    fn path(&self) -> std::borrow::Cow<'static, str> {
+        Cow::Borrowed("/api/0/relays/projectconfigs/?version=4")
+    }
+
+    fn retry() -> bool {
+        false
+    }
+
+    fn priority() -> super::upstream::RequestPriority {
+        RequestPriority::High
+    }
+
+    fn route(&self) -> &'static str {
+        "global_config"
+    }
+}
+
+/// The message for requesting the most recent global config from [`GlobalConfigService`].
+pub struct Get;
+
+/// The message for receiving a watch that subscribes to the [`GlobalConfigService`].
+pub struct Subscribe;
+
+/// A way to get updates of the global config.
+pub enum GlobalConfigManager {
+    /// Returns the most recent global config.
+    Get(relay_system::Sender<Arc<GlobalConfig>>),
+    /// Returns a [`watch::Receiver`] where global config updates will be sent to.
+    Subscribe(relay_system::Sender<watch::Receiver<Arc<GlobalConfig>>>),
+}
+
+impl Interface for GlobalConfigManager {}
+
+impl FromMessage<Get> for GlobalConfigManager {
+    type Response = AsyncResponse<Arc<GlobalConfig>>;
+
+    fn from_message(_: Get, sender: relay_system::Sender<Arc<GlobalConfig>>) -> Self {
+        Self::Get(sender)
+    }
+}
+
+impl FromMessage<Subscribe> for GlobalConfigManager {
+    type Response = AsyncResponse<watch::Receiver<Arc<GlobalConfig>>>;
+
+    fn from_message(
+        _: Subscribe,
+        sender: relay_system::Sender<watch::Receiver<Arc<GlobalConfig>>>,
+    ) -> Self {
+        Self::Subscribe(sender)
+    }
+}
+
+/// Service implementing the [`GlobalConfigManager`] interface.
+///
+/// The service offers two alternatives to fetch the [`GlobalConfig`]:
+/// responding to a [`Get`] message with the config for one-off requests, or
+/// subscribing to updates with [`Subscribe`] to keep up-to-date.
+#[derive(Debug)]
+pub struct GlobalConfigService {
+    config: Arc<Config>,
+    /// Sender of the [`watch`] channel for the subscribers of the service.
+    sender: watch::Sender<Arc<GlobalConfig>>,
```

Comment:
Just a suggestion to make `handle_message` more readable:

```suggestion
    global_config: watch::Sender<Arc<GlobalConfig>>,
```

Reply:
I'm not sure if `global_config` better represents the purpose of this variable. I think it could be confusing with a global config instance when reading the code below. I'm open to changing it but I can't think of better suggestions, maybe "subscriber"?

Reply:
Best to pick a name that makes sense from the point of view of the code that uses this field:

```rust
sender.send(self.sender.borrow().clone());
```

Maybe `global_config_watch`?

Reply:
Sounds great! Renamed in [255f885](https://github.com/getsentry/relay/pull/2320/commits/255f885d8d7cca30f028dca14dd92cbe4e00d29b).